electron/src/tray.js

```diff
@@ -47,21 +46,31 @@ exports.create = function (win, config) {
     const contextMenu = electron.Menu.buildFromTemplate([
         {
             label: 'Show/Hide ' + config.brand,
-            click: toggleWin
+            click: toggleWin,
         },
         {
-            type: 'separator'
+            type: 'separator',
         },
         {
             label: 'Quit',
-            click: function () {
+            click: function() {
                 app.quit();
-            }
-        }
+            },
+        },
     ]);
 
     trayIcon = new Tray(config.icon_path);
     trayIcon.setToolTip(config.brand);
     trayIcon.setContextMenu(contextMenu);
     trayIcon.on('click', toggleWin);
+
+    ipcMain.on('set_badge', function(event, count) {
+        if (count) {
+            trayIcon.setImage(config.icon_path_unread);
+            trayIcon.setToolTip(`${config.brand} — ${count} unread`);
```

Comment:
This text should probably match (and ideally be generated by the same code as) https://github.com/matrix-org/matrix-react-sdk/blob/master/src/components/structures/MatrixChat.js#L1105